src/Aspire.Hosting.Azure/Extensions/AzureStorageExtensions.cs

```diff
@@ -30,6 +32,41 @@ public static IResourceBuilder<AzureStorageResource> AddAzureStorage(this IDistr
                       .WithManifestPublishingCallback(resource.WriteToManifest);
     }
 
+    /// <summary>
+    /// Adds an Azure Storage resource to the application model.This resource can be used to create Azure blob, table, and queue resources.
+    /// </summary>
+    /// <param name="builder">The builder for the distributed application.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="configureResource">Callback to configure the storage account.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<AzureConstructResource> AddAzureConstructStorage(this IDistributedApplicationBuilder builder, string name, Action<ResourceModuleConstruct, StorageAccount>? configureResource = null)
+    {
+        var configureConstruct = (ResourceModuleConstruct construct) =>
+        {
+            var storageAccount = construct.AddStorageAccount(
+                name: name,
+                kind: StorageKind.StorageV2,
+                sku: StorageSkuName.StandardGrs
+                );
+
+            storageAccount.AddOutput(sa => sa.PrimaryEndpoints.BlobUri, "blobEndpoint");
+            storageAccount.AddOutput(sa => sa.PrimaryEndpoints.QueueUri, "queueEndpoint");
+            storageAccount.AddOutput(sa => sa.PrimaryEndpoints.TableUri, "tableEndpoint");
+
+            if (configureResource != null)
+            {
+                configureResource(construct, storageAccount);
```

Comment:
You don't want to pass a single argument and make callers use Parent?

Reply:
I thought about this. However I plan to expose some key properies on the construct, so I need it to be our type, otherwise they'll have to cast to it ... and because the type is not obvious I think the two argument callback is a bit more obvious.

It also works better when we have higher level resources that are logically composed of multiple Azure resources and you want to give an easy way to get to everything that is in the module construct.

We could create a context object for each resource but I felt like that was going a bit too far.